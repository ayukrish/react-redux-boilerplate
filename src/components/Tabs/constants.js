export const tabItems = [
  {
    name: 'Characters',
    id: 'characters',
    routePath: '/'
  },
  {
    name: 'Locations',
    id: 'locations',
    routePath: '/locations'
  },
  {
    name: 'Episodes',
    id: 'episodes',
    routePath: '/episodes'
  }
];
