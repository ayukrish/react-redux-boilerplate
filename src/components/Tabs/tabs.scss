.tabWrapper {
  display: flex;
}

.tab {
  margin: 20px;
  font-size: 18px;
  a {
    color: black;
    text-decoration: none;
    &:hover {
      color: rgb(255, 152, 0);
    }
  }

  &.active {
    a {
      color: rgb(255, 152, 0);
    }
  }
}
