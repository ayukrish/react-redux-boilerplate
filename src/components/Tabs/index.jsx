import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';
import { Link } from 'react-router-dom';
import { tabItems } from './constants';
import styles from './tabs.scss';

const propTypes = {
  history: PropTypes.object.isRequired
};

const defaultProps = {};

class Tabs extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      selectedTab: tabItems[0].id
    };
  }

  componentDidMount() {
    this.setTab();
  }

  componentDidUpdate() {
    this.setTab();
  }

  setTab = () => {
    const { history } = this.props;
    const tabDetails = this.getActiveTab(history.location.pathname);
    if (tabDetails.id !== this.state.selectedTab) {
      this.setState({
        selectedTab: tabDetails.id
      });
    }
  }

  getActiveTab = route => {
    const index = tabItems.findIndex(item => item.routePath === route);
    return index === -1 ? tabItems[0] : tabItems[index];
  }

  getTabItems = () => {
    const { selectedTab } = this.state;
    return tabItems && tabItems.map(item => (
      <div
        key={item.id}
        className={`${styles.tab} ${selectedTab === item.id ? styles.active : ''}`}
      >
        <Link to={item.routePath}>{item.name}</Link>
      </div>
    ));
  }

  render() {
    return (
      <section className={styles.tabWrapper}>
        {this.getTabItems()}
      </section>
    );
  }
}

Tabs.propTypes = propTypes;
Tabs.defaultProps = defaultProps;
export default withRouter(Tabs);
