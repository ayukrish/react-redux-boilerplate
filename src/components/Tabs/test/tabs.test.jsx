import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

// Components
import Tabs from '../index';

describe('Tabs Test Suite', () => {
  it('renders correctly without any data', () => {
    const output = shallow(
      <Tabs.WrappedComponent
        history={{
          location: {
            pathname: ''
          }
        }}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const output = shallow(
      <Tabs.WrappedComponent
        history={{
          location: {
            pathname: '/locations'
          }
        }}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
