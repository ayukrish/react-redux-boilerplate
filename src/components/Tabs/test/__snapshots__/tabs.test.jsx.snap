// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Test Suite renders correctly 1`] = `
<section>
  <div
    className="undefined "
    key="characters"
  >
    <Link
      to="/"
    >
      Characters
    </Link>
  </div>
  <div
    className="undefined undefined"
    key="locations"
  >
    <Link
      to="/locations"
    >
      Locations
    </Link>
  </div>
  <div
    className="undefined "
    key="episodes"
  >
    <Link
      to="/episodes"
    >
      Episodes
    </Link>
  </div>
</section>
`;

exports[`Tabs Test Suite renders correctly without any data 1`] = `
<section>
  <div
    className="undefined undefined"
    key="characters"
  >
    <Link
      to="/"
    >
      Characters
    </Link>
  </div>
  <div
    className="undefined "
    key="locations"
  >
    <Link
      to="/locations"
    >
      Locations
    </Link>
  </div>
  <div
    className="undefined "
    key="episodes"
  >
    <Link
      to="/episodes"
    >
      Episodes
    </Link>
  </div>
</section>
`;
